harmonica/tests/test_filters.py

```diff
@@ -241,8 +250,161 @@ def test_derivative_upward_kernel(sample_fft_grid, order):
+    """
+    # Calculate expected outcome
+    k_easting = 2 * np.pi * sample_fft_grid.freq_easting
+    expected = (k_easting * 1j) ** order
+    # Check if the filter returns the expected output
+    xrt.assert_allclose(
+        expected, derivative_easting_kernel(sample_fft_grid, order=order)
+    )
+
+
+@pytest.mark.parametrize("order", (1, 2, 3))
+def test_derivative_northing_kernel(sample_fft_grid, order):
+    """
+    Check if derivative_northing_kernel works as expected
+    """
+    # Calculate expected outcome
+    k_northing = 2 * np.pi * sample_fft_grid.freq_northing
+    expected = (k_northing * 1j) ** order
+    # Check if the filter returns the expected output
+    xrt.assert_allclose(
+        expected, derivative_northing_kernel(sample_fft_grid, order=order)
+    )
+
+
+@pytest.mark.parametrize("height", (-10, -100, -1000))
+def test_upward_continuation_kernel(sample_fft_grid, height):
+    """
+    Check if upward_continuation_kernel works as expected
+    """
+    # Calculate expected outcome
+    k_easting = 2 * np.pi * sample_fft_grid.freq_easting
+    k_northing = 2 * np.pi * sample_fft_grid.freq_northing
+    expected = np.exp(np.sqrt(k_easting ** 2 + k_northing ** 2) * height)
+    # Check if the filter returns the expected output
+    xrt.assert_allclose(
+        expected, upward_continuation_kernel(sample_fft_grid, height=height)
+    )
+
+
+@pytest.mark.parametrize("wavelength", (10, 100, 1000))
+def test_gaussian_lowpass_kernel(sample_fft_grid, wavelength):
+    """
+    Check if gaussian_lowpass_kernel works as expected
+    """
+    # Calculate expected outcome
+    k_easting = 2 * np.pi * sample_fft_grid.freq_easting
+    k_northing = 2 * np.pi * sample_fft_grid.freq_northing
+    expected = np.exp(
+        -(k_easting ** 2 + k_northing ** 2) / (2 * (2 * np.pi / wavelength) ** 2)
+    )
+    # Check if the filter returns the expected output
+    xrt.assert_allclose(
+        expected, gaussian_lowpass_kernel(sample_fft_grid, wavelength=wavelength)
+    )
+
+
+@pytest.mark.parametrize("wavelength", (10, 100, 1000))
+def test_gaussian_highpass_kernel(sample_fft_grid, wavelength):
+    """
+    Check if gaussian_highpass_kernel works as expected
+    """
+    # Calculate expected outcome
+    k_easting = 2 * np.pi * sample_fft_grid.freq_easting
+    k_northing = 2 * np.pi * sample_fft_grid.freq_northing
+    expected = 1 - np.exp(
+        -(k_easting ** 2 + k_northing ** 2) / (2 * (2 * np.pi / wavelength) ** 2)
+    )
+    # Check if the filter returns the expected output
+    xrt.assert_allclose(
+        expected, gaussian_highpass_kernel(sample_fft_grid, wavelength=wavelength)
+    )
+
+
+def test_reduction_to_pole_kernel(sample_fft_grid, i=60, d=45, im=45, dm=50):
+    """
+    Check if reduction_to_pole_kernel works as expected
+    """
+    # Transform degree to rad
+    [i, d] = np.deg2rad([i, d])
+
+    if dm is None or im is None:
+        [im, dm] = [i, d]
+    else:
```

Comment:
These lines aren't needed in the test, since the `im` and `dm` always have values that are not None.

Reply:
yup